### PR TITLE
check if ClusterMachineApproverDown UsingDeprecatedAPIExtensionsV1Beta1 alerts are not firing

### PR DESCRIPTION
**Pull request by paulfantom**
Follow-up to #24019.

/cc @LiliC @simonpasquier @s-urbaniak

### CODE REVIEW

File: test/extended/prometheus/prometheus.go [hunk 1]
```diff
@@ -210,7 +210,7 @@ var _ = g.Describe("[Feature:Prometheus][Conformance] Prometheus", func() {
 			tests := map[string]bool{
 				// Checking Watchdog alert state is done in "should have a Watchdog alert in firing state".
 				// FIXME(paulfantom): ClusterMachineApproverDown and UsingDeprecatedAPIExtensionsV1Beta1 shouldn't be firing!
```
Comment (simonpasquier) [inline test/extended/prometheus/prometheus.go]: this line can be removed

**Comment by paulfantom** [comment]
@simonpasquier removed comment.

Putting this on hold as `UsingDeprecatedAPIExtensionsV1Beta1` is firing constantly. BZ for this issue is created at https://bugzilla.redhat.com/show_bug.cgi?id=1768417

/hold

**Comment by tnozicka** [comment]
https://github.com/openshift/origin/pull/24089 merged
/retest

**Comment by tnozicka** [comment]
https://github.com/openshift/cluster-kube-apiserver-operator/pull/639 merged
/retest

**Comment by tnozicka** [comment]
/hold cancel
(the 2 PRs above fixed the e2e here)

**Comment by tnozicka** [comment]
gcp flake
/retest

**Comment by tnozicka** [comment]
/hold
seem to be a flake depending on when the test is run

**Comment by paulfantom** [comment]
@tnozicka failure looks legitimate to me. Log message says:
```
ALERTS{alertname=\"UsingDeprecatedAPIExtensionsV1Beta1\", alertstate=\"firing\", client=\"kubectl/v0.0.0 (linux/amd64) kubernetes/$Format\", code=\"200\", component=\"apiserver\", contentType=\"application/json\", endpoint=\"https\", group=\"extensions\", instance=\"10.0.0.4:6443\", job=\"apiserver\", namespace=\"default\", resource=\"deployments\", scope=\"namespace\", service=\"kubernetes\", severity=\"warning\", verb=\"DELETE\", version=\"v1beta1\"} => 1
...
```

**Comment by tnozicka** [comment]
sure, I should fail, yet I see 2 run where it was green https://prow.svc.ci.openshift.org/pr-history/?org=openshift&repo=origin&pr=24071

I am working on a fix, tracing down some more offenders

**Comment by paulfantom** [comment]
> sure, I should fail, yet I see 2 run where it was green

Confirmed. Looks like we need to rewrite those tests to be running during a whole test suite run ASAP.

**Comment by tnozicka** [comment]
/test e2e-aws-serial
/test e2e-gcp

to see if it doesn't fail when scheduled in a different slot, I think there was still one last client I needed to trace and fix

**Comment by paulfantom** [comment]
Rebased to resolve merge conflict

**Comment by paulfantom** [comment]
@tnozicka what is the status of fixing `UsingDeprecatedAPIExtensionsV1Beta1` alerts? Asking because I just started `4.3.0-0.ci-2019-11-22-122829` and got 7 alerts with warning type and this isn't something we would like to ship to customers.

**Comment by paulfantom** [comment]
Rebased after changes from #24276

**Comment by LiliC** [comment]
@tnozicka FYI in the latest nightly 4.4 cluster the `UsingDeprecatedAPIExtensionsV1Beta1` is still firing, when will this issue be resolved do you know?

**Comment by LiliC** [comment]
Also the `UsingDeprecatedAPIExtensionsV1Beta1 ` is firing 20 times on my nightly 4.4 cluster, this should be amended to group those and only fire once, not sure how the alert is created?

**Comment by LiliC** [comment]
/hold

Until the alert is fixed, so we don't keep retrying.

**Comment by tnozicka** [comment]
looks like people managed to merge new offenders in the meantime
one fix is here https://github.com/openshift/origin/pull/24305

**Comment by LiliC** [comment]
@tnozicka Yeah, guess this is why we need to enable them ASAP. Have you considered adding an e2e test to origin instead of an alert for this?  As this will be firing on user clusters and I assume a lot of confusion will happen as users can't do anything about this from their side.

**Comment by soltysh** [comment]
With the updated k8s where we **DO NOT** enable `extensions/v1beta1` every tests that uses it should fail so this is free to merge.
/hold cancel

**Pull request merged**